### PR TITLE
Change how ROS 2 packages are included.

### PR DESCRIPTION
**Pull request by michalpelka**
Include cmake configs from the ROS 2 package instead of building lib and include paths.
Added function `target_depends_on_ros2_package` that allows specifying version and optionality of the ROS 2 package.
Requested minimum version of control_toolbox to be 2.2.0.
Addresses issues :
#198

### CODE REVIEW

File: Gems/ROS2/Code/ros2_target_depends.cmake [hunk 1]
```diff
@@ -3,29 +3,26 @@
...
         message(DEBUG "Processing package: ${_package}")
-        find_package(${_package} REQUIRED)
-        list(APPEND _ros2_package_libraries "${${_package}_LIBRARIES}")
-        list(APPEND _ros2_include_directories "${${_package}_INCLUDE_DIRS}")
+        target_depends_on_ros2_package(${TARGET_NAME} ${_package} REQUIRED)
     endforeach()
-    target_include_directories(${TARGET_NAME} PUBLIC ${_ros2_include_directories})
-    target_link_libraries(${TARGET_NAME} PUBLIC ${_ros2_package_libraries})
 endfunction()
+
```
Comment (adamdbrw) [inline Gems/ROS2/Code/ros2_target_depends.cmake:25]: nitpick: too many newlines here

**Review by adamdbrw — COMMENTED** [review]
One question: we depend on a specific version of control toolbox package, but is this reflected in the installation instructions / process?

File: Gems/ROS2/Code/ros2_target_depends.cmake [hunk 2]
```diff
@@ -3,29 +3,26 @@
...
+function(target_depends_on_ros2_package TARGET_NAME )
+    list (GET ARGN 0 _package)
+    find_package(${ARGN})
+    include(${${_package}_DIR}/${_package}Config.cmake OPTIONAL)
+    include(${${_package}_DIR}/${_package}config.cmake OPTIONAL)
+    include(${${_package}_DIR}/${_package}-config.cmake OPTIONAL)
+    include(${${_package}_DIR}/${_package}-Config.cmake OPTIONAL)
+    if( ${${_package}_FOUND} )
+        message(DEBUG "Package ${_package} was found (${${_package}_DIR}) version ${${_package}_VERSION} targets : ${${_package}_TARGETS}")
+        target_link_libraries(${TARGET_NAME} PUBLIC ${${_package}_TARGETS})
```
Comment (spham-amzn) [inline Gems/ROS2/Code/ros2_target_depends.cmake:15]: nit: Should add an `else()` clause here to maybe debug print that the package was not found (or is it an error?)

File: Gems/ROS2/Code/ros2_target_depends.cmake [hunk 3]
```diff
@@ -3,29 +3,26 @@
...
-    set(_ros2_include_directories)
-    set(_ros2_package_libraries)
-    foreach(_ros2_packages_path IN LISTS _ament_prefix_path)
-        string(REPLACE ":" ";" _ros2_packages_path ${_ros2_packages_path})
-        list(APPEND _ros2_library_directories "${_ros2_packages_path}/lib")
-        list(APPEND _ros2_include_directories "${_ros2_packages_path}/include")
-    endforeach()
+function(target_depends_on_ros2_packages TARGET_NAME)
     foreach(_package IN LISTS ARGN)
         message(DEBUG "Processing package: ${_package}")
```
Comment (spham-amzn) [inline Gems/ROS2/Code/ros2_target_depends.cmake:21]: Probably don't need this message if we are debug printing in the `target_depends_on_ros2_package` function

**Review by spham-amzn — APPROVED** [review]

File: Gems/ROS2/Code/ros2_target_depends.cmake (hunk 3 above)
Comment (michalpelka) [reply Gems/ROS2/Code/ros2_target_depends.cmake]: Thanks, removed.

File: Gems/ROS2/Code/ros2_target_depends.cmake (hunk 2 above)
Comment (michalpelka) [reply Gems/ROS2/Code/ros2_target_depends.cmake:12]: I've added error.

**Comment by michalpelka** [comment]
Thanks for reviewing @spham-amzn. Due to your questions, I've found out how to avoid building technological debt inside this PR. I was trying to include multiple cmake files with OPTIONAL, which was sub-optimal.
The urdf is not an ament package, it is an ordinary cmake package, so I've moved it to the main CMakeList.txt and used ordinary 'find_package'

**Review by hultonha — APPROVED** [review]
LGTM :+1:

File: Gems/ROS2/Code/ros2_target_depends.cmake [hunk 4]
```diff
@@ -3,29 +3,20 @@
...
-        return()
+function(target_depends_on_ros2_package TARGET_NAME )
+    list (GET ARGN 0 _package)
+    find_package(${ARGN})
+    include(${${_package}_DIR}/${_package}Config.cmake OPTIONAL)
+    if( ${${_package}_FOUND_AMENT_PACKAGE} )
+        message(DEBUG "Package ${_package} was found (${${_package}_DIR}) version ${${_package}_VERSION} targets : ${${_package}_TARGETS}")
+        target_link_libraries(${TARGET_NAME} PUBLIC ${${_package}_TARGETS})
+    else()
+        message(FATAL_ERROR "Package ${_package} was found (${${_package}_DIR}), but package is not Ament package. ")
```
Comment (adamdbrw) [inline Gems/ROS2/Code/ros2_target_depends.cmake:14]: not an Ament

File: Gems/ROS2/Code/ros2_target_depends.cmake [hunk 5]
```diff
@@ -3,29 +3,20 @@
...
 
-function(target_depends_on_ros2_packages TARGET_NAME)
-    find_package(ROS2 MODULE)
-    if (NOT ROS2_FOUND)
-        return()
+function(target_depends_on_ros2_package TARGET_NAME )
+    list (GET ARGN 0 _package)
+    find_package(${ARGN})
+    include(${${_package}_DIR}/${_package}Config.cmake OPTIONAL)
+    if( ${${_package}_FOUND_AMENT_PACKAGE} )
```
Comment (adamdbrw) [inline Gems/ROS2/Code/ros2_target_depends.cmake:10]: formatting

File: Gems/ROS2/Code/CMakeLists.txt [hunk 6]
```diff
@@ -44,7 +45,13 @@ ly_add_target(
             Gem::PhysX.Static
 )
 
-target_depends_on_ros2_packages(${gem_name}.Static rclcpp builtin_interfaces std_msgs sensor_msgs nav_msgs urdfdom tf2_ros ackermann_msgs gazebo_msgs control_toolbox)
+find_package(urdfdom)
+target_link_libraries(${gem_name}.Static PUBLIC urdfdom::urdfdom_model)
```
Comment (adamdbrw) [inline Gems/ROS2/Code/CMakeLists.txt:49]: Editor target

Comment (michalpelka) [reply Gems/ROS2/Code/CMakeLists.txt]: Nice catch @adamdbrw ! Adjusted and tested.

**Review by adamdbrw — APPROVED** [review]

**Pull request merged**